fix: make Tags mandatory in ForwardConditions to match Rust backend

sapkota-aayush opened the pull request:
Fixes #2809

### What this PR does / why we need it
Fixes the inconsistent contract between Go and Rust backends for edge conditions tags. Previously, the Go backend allowed empty `conditions: {}` (with nil Tags) while the Rust backend required Tags to be present, causing validation failures when switching between backends.

### Related issues
Fixes #2809

### Testing
- **Unit tests**: All 227 unit tests pass, including 7 conditional forwarding test cases
- **Validation logic**: Updated to work with mandatory Tags field
- **Generated code**: Updated protobuf and deepcopy files to match new struct definition
- **Commands run**: `go test ./pkg/reconciler/validator/... -v`

### Special notes for reviewers
This is a breaking change that makes the Go backend behavior consistent with Rust. Users who previously used empty `conditions: {}` will now need to provide meaningful Tags. The change enforces better API design by requiring meaningful conditions when specified.

**Changes made:**
- Remove pointer from Tags field in ForwardConditions struct
- Change protobuf annotation from opt to req for Tags  
- Update validation logic to remove nil check
- Fix unit tests to use non-pointer syntax
- Update generated protobuf and deepcopy files
<img width="508" height="1036" alt="Test1Go" src="https://github.com/user-attachments/assets/4552623b-90b5-413a-8515-ad9cf9a53bfb" />
<img width="592" height="1032" alt="test2Go" src="https://github.com/user-attachments/assets/d07044f3-626c-4106-a162-83a8a6f7da67" />

vigith commented on docs/development/docker-optimization.md:
why are these added?

vigith reviewed (changes requested):
A lot of  non related new files have also been added to this PR, please remove those.

sapkota-aayush commented:
@vigith  I've cleaned up the PR and removed all unrelated files. The PR now contains only the Tags field fix and regenerated codegen files. All unit tests pass including the conditional forwarding tests. Ready for review.
The failing E2E tests (idle-source-e2e and sideinputs-e2e) appear to be unrelated to the Tags field changes and are likely pre-existing CI issues.

vigith commented:
thank you @sapkota-aayush, PR looks good to.

KeranYang commented:
Thank you @sapkota-aayush

vigith commented:
@sapkota-aayush i don't think you pulled in the latest main, that is the reason for showing "Conflicts"
